bokehjs/src/lib/models/widgets/sliders/categorical_slider.ts

```diff
@@ -16,28 +16,45 @@ export class CategoricalSliderView extends AbstractSliderView<string> {
   }
 
   protected _calc_to(): SliderSpec<string> {
-    const {categories} = this.model
+    const {categories, value} = this.model
+    let start_idx = categories.indexOf(String(value))
```

Comment:
When appropriate types are restored, then this `String(...)` coercion won't be needed anymore.